langchain4j-alloydb-pg/src/main/java/dev/langchain4j/store/embedding/alloydb/AlloyDBEmbeddingStore.java

```diff
@@ -89,7 +93,28 @@ public EmbeddingSearchResult<TextSegment> search(EmbeddingSearchRequest request)
 
     @Override
     public void removeAll(Collection<String> ids) {
-        throw new UnsupportedOperationException("Not supported yet.");
+        if (ids == null || ids.isEmpty()) {
+            throw new IllegalArgumentException("ids must not be null or empty");
+        }
+
+        String placeholders = "?";
+        for (int p = 1; p < ids.size(); p++) {
+            placeholders += ", ?";
+        }
+
+        String query = "DELETE FROM \"" + schemaName + "\".\"" + tableName + "\" WHERE \"" + idColumn + "\" IN (" + placeholders + ")";
+        try (PreparedStatement preparedStatement = engine.getConnection().prepareStatement(query)) {
+            int i = 1;
+            for (String id : ids) {
+                preparedStatement.setString(i++, id);
+            }
+            preparedStatement.executeUpdate();
```

Comment:
The string format here works since it cleans up the code nicely
```suggestion
        String query = String.format("DELETE FROM \"%s\".\"%s\" WHERE %s IN (?)", schemaName, tableName, idColumn)
        try (Connection conn = pool.getConnection()) {
            try (PreparedStatement preparedStatement = conn.prepareStatement(query)) {
                Array array = connection.createArrayOf("uuid", ids.stream().map(UUID::fromString).toArray());
                preparedStatement.setArray(1, array);
                preparedStatement.executeUpdate();
            }
```